pkg/jobs/jobspb/jobs.proto

```diff
@@ -1191,14 +1191,37 @@ message ResolvedSpans {
 message ChangefeedProgress {
   reserved 1;
 
-  // Checkpoint describes a list of spans that the changefeed has progressed to
-  // at or above the specified timestamp.  This is used for situations where
-  // some spans are significantly behind the rest, such as during backfills or
-  // if specific spans are having issues progressing.  On restart, the
-  // changefeed will forward the specified spans to the specified timestamp.
+  // Checkpoint contains a list of spans that are above the changefeed's
+  // highwater (overall resolved) timestamp. In an earlier version, it consisted
+  // of a single timestamp and a list of spans. This single timestamp was the
+  // min resolved timestamp among the list of spans. The current version of the
+  // checkpoint is a "map" from timestamps to lists of spans, where each entry
+  // is a unique resolved timestamp and a list of associated spans.
+  //
+  // On restart, the changefeed can restore the progress of spans in the
+  // checkpoint to its corresponding resolved timestamp, which will be higher
+  // than the overall resolved timestamp and thus allow us to do less work.
+  // This is especially useful during backfills or if some spans are lagging.
   message Checkpoint {
```

Comment:
Could you rename the `Checkpoint` message `DeprecratedCheckpoint`, and pull the `TiemstampSpansMap` message into its own first class proto message called `CheckpointMap` or something? This feels cleaner because: - The new and old checkpoint fields do not interact, so i don't think they need to be so tightly coupled in the current `Checkpoint` proto.
- once you truly deprecate those first two fields, there will be an unnecessary level of hierarchy in the checkpoint proto-- the current `timestamp_spans` field would be the only field in `Checkpoint`.

Reply:
Done. I kept the first class proto message name as `TimestampSpansMap` but I renamed the field inside `ChangefeedProgress` to `span_level_checkpoint`.

Reply:
Discussed offline and we decided to not rename the message or fields and instead mark the message as deprecated. (The fields were already marked as deprecated.)